www/js/app.js

```diff
@@ -49,6 +49,18 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
      */
     var selectedArchive = null;
     
+    /**
+     * A global parameter object for storing variables that need to be remembered between page loads
+     * or across different functions
+     * 
+     * @type Object
+     */
+    var params = {};
```

Comment:
I can't make my mind about the use of this `params` object.
On which criteria would you put some variables in it, instead of declaring them individually? How different are they from some other variable like contentInjectionMode, lastContentInjectionMode, serviceWorkerRegistration etc?

Reply:
@mossroy They are not really different. It just simplified the back-porting of the code from Kiwix JS Windows. The architecture I adopted there, first suggested by sharun-s I believe, was to declare a global params object in `init.js` and populate it from cookie at startup, set UI checkboxes, etc., or set default parameters which the user can then override in the UI. I have about 20 parameters that control different things about the state of the app (not all stored in the cookie by any means), and it simplifies things greatly to have them all kept together in the one object, and in one spot which the developer can easily manipulate (a bit like an old .ini file). There's an argument for tidying up and centralizing the use of such parameters in Kiwix JS, and things like `params['lastContentInjectionMode']` would be prime candidates for such tidying. However, I recognize that this is importing a bit of "foreign" code, and if you'd rather not go down that route, at least for now, it's fine, I can easily rename the two params I use in this PR. Just let me know if you'd prefer me to rename to standard variables.

Reply:
I've added issue #469 .